basic_auth.go

```diff
@@ -21,8 +21,7 @@ type basicAuth struct {
 // default error handler if you wish to serve a custom template/response.
 type AuthOptions struct {
 	Realm               string
-	User                string
-	Password            string
+	AuthFunc            func(string, string) bool
```

Comment:
This will unfortunately break backwards compatibility ("the API") for other users, and I don't think it's worth it for a fairly minor feature.